### ISSUE TITLE
host_redirect causes config error with 0.50 (v2 api)

### ISSUE THREAD

**Issue by kellycampbell** [datawire/ambassador#1067]
**Describe the bug**
Using a `host_redirect` mapping causes an envoy api config error.

```
invalid value oneof field 'action' is already set. Cannot set 'redirect' for type oneof
```

Mapping annotation was like this:
```
      apiVersion: ambassador/v1
      kind: Mapping
      name: test-redirect
      host: host.mydomain.com
      host_redirect: true
      service: https://otherhost.mydomain.com
      prefix: /
```

Without the `prefix: /` line, it gets a different error:
```
<RichStatus BAD error=not a valid Mapping: 'prefix' is a required property
```

Full error:
```
2018-12-27 00:50:16.116][000038][warning][config] [bazel-out/k8-opt/bin/source/common/config/_virtual_includes/grpc_mux_subscription_lib/common/config/grpc_mux_subscription_impl.h:70] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener ir.listener: Unable to parse JSON as proto (INVALID_ARGUMENT:(route_config.virtual_hosts[0].routes[18]): invalid value oneof field 'action' is already set. Cannot set 'redirect' for type oneof):
```

**Expected behavior**

Expected valid envoy config 

**Versions:**
 - Ambassador: 0.50.0-rc2
 - Kubernetes environment AWS kops 1.11-beta.1
 - Version 1.11.6

**Comment by kflynn** [comment]
Fixed in #1073.

**Comment by kflynn** [comment]
Actually, leaving this open until we release RC3.
